lib-extra/src/main/java/com/diffplug/spotless/extra/config/SemanticVersion.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2016 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.extra.config;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.Objects;
+
+/** Semantic version configuration */
+class SemanticVersion implements Serializable, Comparable<SemanticVersion> {
+
+	public static final String FORMAT = "<major>[.minor][.patch]";
+	private static final int MAJOR = 0;
+	private static final int MINOR = 1;
+	private static final int PATCH = 2;
+
+	// Not used, only the serialization output is required to determine whether the object has changed
+	private static final long serialVersionUID = 1L;
+
+	private final int[] version;
```

Comment:
Imho 3 separate `int` fields would represent the state in this class a bit cleaner

Reply:
Don't know, had it in the beginning like that and found the code ugly. Now I find it less ugly :wink:. In the end a semantic version always depends on three digits, which needs to be processed in order. This is a principal used in many places. For readability, this principal could be sub-classed, where the sub-class has the three values as separated members but provide also list functionality. But I found that a little bit too verbose.